File: max_heap_property.java
import java.util.*; 
import java.io.*;

class Main {
  public static String MaxHeapChecker(int[] arr) {
    String word = "";
    boolean max_property = false;
    int root = arr[0];  
    for(int i = 0;i<arr.length;++i){
      root = arr[i-1];
      if(arr[(2*i)-1]<root && arr[((2*i)+1)-1]<root){ 
        max_property = true;  
      }
      else{
        max_property = false;
      }
      if(max_property){
        return "max";
      }
      else{
        for(int j = 0;j<arr.length;++j){
        //word+=to_string(arr[j]);
        }
      }
      return word;
    }  
    // code goes here  
    return word;
  }

  public static int[] parseInput(String input){
    String[] stringArr = input.split(","); // Assuming input is comma-separated
    int[] intArr = new int[stringArr.length];
    for (int i = 0; i < stringArr.length; i++) {
      intArr[i] = Integer.parseInt(stringArr[i].trim());
      }
    return intArr;
  }

  public static void main (String[] args) {  
    // keep this function call here     
    Scanner s = new Scanner(System.in);
    //System.out.print(MaxHeapChecker(s.nextLine()))
    String input =s.nextLine();
    int [] arr = parseInput(input); 
    MaxHeapChecker(arr);
  }

}

File: max_heap_property.cpp
#include <iostream>
#include <string>
using namespace std;

string MaxHeapChecker(int arr[], int arrLength) {
  string word;
  int counter = 0;
  // code goes here
  int root = 0;
  int pulled = 0;
  bool max_property = false;  
  for(int i = 1;i<arrLength and 2*i<arrLength;++i){ //heap-index:from 1,2,3...->0,1,2... minus 1
      root = arr[i-1]; //floor(i/2)-1
      if(arr[(2*i)-1]>root or arr[((2*i)+1)-1]>root){ //1,2
        max_property = false;  //first children: (2*1)-1,(2*1+1)-1
         ++counter; //how many broken max_properties
         if(arr[(2*i)-1]>root){
          pulled = (2*i)-1;
          if(word.empty()){
            word+=to_string(arr[pulled]);
          }
          else{
            word+=",";
            word+=to_string(arr[pulled]);
          }
         }
         if(arr[((2*i)+1)-1]>root){
           pulled = ((2*i)+1)-1;
          if(word.empty()){
            word+=to_string(arr[pulled]);
          }
          else{
            word+=",";
            word+=to_string(arr[pulled]);
          }
         }
         
      
      }
      else{
        //cout<< "index" << 2*i << endl;
       
        max_property = true;

      }
  }
  if(max_property and counter == 0){ //no broken max properties
    return "max";
  }
  else{
    for(int j =0;j<arrLength;++j){
      //word+=to_string(arr[j]);
      //word = to_string(pulled);
    }
    return word;
  }
  

}

int main(void) { 
   
  // keep this function call here
  int A[] = coderbyteInternalStdinFunction(stdin);
  int arrLength = sizeof(A) / sizeof(*A);
  cout << MaxHeapChecker(A, arrLength);
  return 0;
    
}
